LayoutTests/accessibility/mac/text-operation/text-operation-capitalize.html

```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
+<html>
+<head>
+<script src="../../../resources/accessibility-helper.js"></script>
+<script src="../../../resources/js-test.js"></script>
+</head>
+<body>
+
+<div id="test-content">
+    <p contenteditable="true" id="text">The quick <span id="target1">brown fo</span>x jumps ov<span id="target2">er the la</span>zy dog.</p>
+    <p contenteditable="true" id="text2">TEXT2: The quick brown fox jumps over the lazy dog.</p>
+    <p contenteditable="true" id="text3">TEXT3: The quick brown fox jumps over the lazy dog.</p>
+</div>
+
+<script>
+var output = "Tests that a Capitalize text operation transforms text to capitalized as expected.\n\n";
+
+async function getMarkerRangeInElementById(id, startIndex, endIndex) {
+    const root = accessibilityController.rootElement.childAtIndex(0);
+    const selection = window.getSelection();
+    selection.removeAllRanges();
+
+    await waitFor(() => {
+        const selectedRange = root.selectedTextMarkerRange();
+        return root.textMarkerRangeLength(selectedRange) == 0;
+    });
+
+    const range = document.createRange();
+    const element = document.getElementById(id);
+    range.setStart(element.firstChild, startIndex);
+    range.setEnd(element.firstChild, endIndex);
+    selection.addRange(range);
+
+    let selectedRange;
+    await waitFor(() => {
+        selectedRange = root.selectedTextMarkerRange();
+        return root.textMarkerRangeLength(selectedRange) > 0;
+    });
+
+    return selectedRange;
+}
+
+async function waitForNotification(accessibilityElement, notification, operation) {
+    var reached = false;
+    function listener(receivedNotification) {
+        if (receivedNotification == notification) {
+            reached = true;
+            accessibilityElement.removeNotificationListener(listener);
+        }
+    }
+
+    accessibilityElement.addNotificationListener(listener);
+    await operation();
+    await waitFor(() => { return reached; });
```

Comment:
I ran into flakiness when using a Promise directly, where the element's stringValue hadn't updated. A waitFor seems more reliable (I ran it 1000 times without failure).

Please let me know I should take a different approach, like waiting on the expected value. Wanted to implement something along the lines of this https://github.com/WebKit/WebKit/pull/33440#discussion_r1754776076